sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamFileCatalog.scala

```diff
@@ -54,6 +54,6 @@ class StreamFileCatalog(sqlContext: SQLContext, path: Path) extends FileCatalog
   override def refresh(): Unit = {}
 
   override def allFiles(): Seq[FileStatus] = {
-    fs.listStatus(metadataLog.get(None, None).flatMap(_._2).map(new Path(_)))
+    fs.listStatus(metadataLog.allFiles().map(f => new Path(f.path)))
```

Comment:
> We should directly construct the file status here, instead of making another remote call.

Call the `FileStatus`'s constructor? Do we need to store other fields of FileStatus in the log?